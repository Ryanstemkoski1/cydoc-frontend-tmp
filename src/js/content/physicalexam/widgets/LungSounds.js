import React, {Component} from 'react'
import { Table} from 'semantic-ui-react'
import LungSoundsButtons from './LungSoundsButtons'
import HPIContext from '../../../contexts/HPIContext'

export default class LungSounds extends Component {

    static contextType = HPIContext 

    render = () => {
        var lung_map = []
        const lung_lobes = Object.keys(this.context["Physical Exam"]["Lungs"])
        for (var lung_index = 0; lung_index < lung_lobes.length/2; lung_index ++) {
            lung_map.push(
                <Table.Row key={lung_lobes[2*lung_index]}>
                    <Table.Cell verticalAlign='top'> <div style={{marginBottom: 5}}> {lung_lobes[2*lung_index]} </div>
                        <LungSoundsButtons key={lung_lobes[2*lung_index]} lung_lobe={lung_lobes[2*lung_index]}/> 
                    </Table.Cell>
                    <Table.Cell verticalAlign='top'> <div style={{marginBottom: 5}}> {lung_lobes[2*lung_index+1]} </div> 
                        <LungSoundsButtons key={lung_lobes[2*lung_index+1]} lung_lobe={lung_lobes[2*lung_index+1]} position={'top right'}/> 
                    </Table.Cell>
                </Table.Row>
            )
        }
        return (
            <Table celled fixed>
              <Table.Body>
                {lung_map}
              </Table.Body>
              </Table>
              )
        }
    }
