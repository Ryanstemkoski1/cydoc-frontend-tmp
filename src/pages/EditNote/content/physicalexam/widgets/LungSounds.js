import React, {Component} from 'react'
import { Table} from 'semantic-ui-react'
import LungSoundsButtons from './LungSoundsButtons'
import HPIContext from 'contexts/HPIContext.js'

export default class LungSounds extends Component {

    static contextType = HPIContext 

    render = () => {
        var lung_map = []
        const lung_lobes = Object.keys(this.context["Physical Exam"].widgets["Lungs"])
        for (var lung_index = 0; lung_index < lung_lobes.length/2; lung_index ++) {
            lung_map.push(
                <Table.Row key={lung_lobes[2*lung_index]}>
                    <Table.Cell verticalAlign='top'> <div style={{marginBottom: 5}}> {lung_lobes[2*lung_index]} </div>
                        <LungSoundsButtons key={lung_lobes[2*lung_index]} lung_lobe={lung_lobes[2*lung_index]} position={'bottom center'}/> 
                    </Table.Cell>
                    <Table.Cell verticalAlign='top'> <div style={{marginBottom: 5}}> {lung_lobes[2*lung_index+1]} </div> 
                        <LungSoundsButtons key={lung_lobes[2*lung_index+1]} lung_lobe={lung_lobes[2*lung_index+1]} position={'bottom center'}/> 
                    </Table.Cell>
                </Table.Row>
            )
        }
        return (
            <div> 
            <Table celled fixed>
              <Table.Body>
                {lung_map}
              </Table.Body>
              </Table>
              </div>
              )
        }
    }
