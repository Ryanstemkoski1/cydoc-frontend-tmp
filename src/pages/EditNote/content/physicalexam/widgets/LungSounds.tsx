import React, { Component } from 'react';
import { Table } from 'semantic-ui-react';
import LungSoundsButtons from './LungSoundsButtons';
import './TableCSS.css';

export default class LungSounds extends Component {
    render = () => {
        return (
            <div>
                <Table celled unstackable>
                    <Table.Body>
                        <Table.Row key='upperLobe'>
                            <Table.Cell verticalAlign='top'>
                                {' '}
                                <div> Left Upper Lobe</div>
                                <LungSoundsButtons
                                    key='leftUpperLobe'
                                    lungLobe='leftUpperLobe'
                                />
                            </Table.Cell>
                            <Table.Cell verticalAlign='top'>
                                {' '}
                                <div> Right Upper Lobe</div>
                                <LungSoundsButtons
                                    key='rightUpperLobe'
                                    lungLobe='rightUpperLobe'
                                />
                            </Table.Cell>
                        </Table.Row>
                        <Table.Row key='middleLobe'>
                            <Table.Cell verticalAlign='top'>
                                {' '}
                                <div> Lingula </div>
                                <LungSoundsButtons
                                    key='lingula'
                                    lungLobe='lingula'
                                />
                            </Table.Cell>
                            <Table.Cell verticalAlign='top'>
                                {' '}
                                <div> Right Middle Lobe</div>
                                <LungSoundsButtons
                                    key='rightMiddleLobe'
                                    lungLobe='rightMiddleLobe'
                                />
                            </Table.Cell>
                        </Table.Row>
                        <Table.Row key='lowerLobe'>
                            <Table.Cell verticalAlign='top'>
                                {' '}
                                <div> Left Lower Lobe</div>
                                <LungSoundsButtons
                                    key='leftLowerLobe'
                                    lungLobe='leftLowerLobe'
                                />
                            </Table.Cell>
                            <Table.Cell verticalAlign='top'>
                                {' '}
                                <div> Right Lower Lobe</div>
                                <LungSoundsButtons
                                    key='rightLowerLobe'
                                    lungLobe='rightLowerLobe'
                                />
                            </Table.Cell>
                        </Table.Row>
                    </Table.Body>
                </Table>
            </div>
        );
    };
}
