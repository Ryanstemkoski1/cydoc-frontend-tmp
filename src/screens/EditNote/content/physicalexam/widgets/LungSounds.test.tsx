import { shallow } from 'enzyme';
import React from 'react';
import LungSounds from './LungSounds';
import { describe, expect, it } from 'vitest';

describe('LungSounds', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<LungSounds />);
        expect(wrapper).toBeTruthy();
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<LungSounds />);
        expect(wrapper).toMatchSnapshot();
    });
});
