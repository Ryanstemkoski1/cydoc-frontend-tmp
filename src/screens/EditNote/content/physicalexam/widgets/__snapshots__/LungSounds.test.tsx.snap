// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LungSounds > matches snapshot 1`] = `ShallowWrapper {}`;

exports[`LungSounds matches snapshot 1`] = `
<div
  className="full-view-table"
>
  <Table
    as="table"
    celled={true}
    unstackable={true}
  >
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="upperLobe"
      >
        <TableCell
          as="td"
          verticalAlign="top"
        >
           
          <div>
             Left Upper Lobe
          </div>
          <Connect(LungSoundsButtons)
            key="leftUpperLobe"
            lungLobe="leftUpperLobe"
          />
        </TableCell>
        <TableCell
          as="td"
          verticalAlign="top"
        >
           
          <div>
             Right Upper Lobe
          </div>
          <Connect(LungSoundsButtons)
            key="rightUpperLobe"
            lungLobe="rightUpperLobe"
          />
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="middleLobe"
      >
        <TableCell
          as="td"
          verticalAlign="top"
        >
           
          <div>
             Lingula 
          </div>
          <Connect(LungSoundsButtons)
            key="lingula"
            lungLobe="lingula"
          />
        </TableCell>
        <TableCell
          as="td"
          verticalAlign="top"
        >
           
          <div>
             Right Middle Lobe
          </div>
          <Connect(LungSoundsButtons)
            key="rightMiddleLobe"
            lungLobe="rightMiddleLobe"
          />
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
        key="lowerLobe"
      >
        <TableCell
          as="td"
          verticalAlign="top"
        >
           
          <div>
             Left Lower Lobe
          </div>
          <Connect(LungSoundsButtons)
            key="leftLowerLobe"
            lungLobe="leftLowerLobe"
          />
        </TableCell>
        <TableCell
          as="td"
          verticalAlign="top"
        >
           
          <div>
             Right Lower Lobe
          </div>
          <Connect(LungSoundsButtons)
            key="rightLowerLobe"
            lungLobe="rightLowerLobe"
          />
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</div>
`;
